nautobot/cloud/models.py

```diff
@@ -0,0 +1,43 @@
+from django.db import models
+
+from nautobot.core.constants import CHARFIELD_MAX_LENGTH
+from nautobot.core.models.generics import PrimaryModel
+from nautobot.extras.utils import extras_features
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class CloudAccount(PrimaryModel):
+    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, help_text="The name of this Cloud Account.", unique=True)
+    description = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True)
+    account_number = models.CharField(
+        max_length=CHARFIELD_MAX_LENGTH, help_text="The account identifier of this Cloud Account."
+    )
+    provider = models.ForeignKey(
+        to="dcim.Manufacturer",
+        on_delete=models.PROTECT,
+        related_name="cloud_accounts",
+    )
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        default=None,
+        blank=True,
+        null=True,
+    )
+
+    class Meta:
+        ordering = ["provider", "name", "account_number"]
+
+    def __str__(self):
+        return f"{self.provider}: {self.name} - {self.account_number}"
```

Comment:
Personally I'd prefer `str` to just be the unique name of the instance, the more verbose string can be reserved for `display`.